fetcher/fetcher.go

```diff
@@ -210,12 +202,13 @@ func (f *Fetcher) Stop() {
 	})
 }
 
-// Wait until all the workers are done and return any errors
+// Wait returns as soon as all enqueued items are fetched and
+// returns any errors.
+//
+// You should not call [Lookup] after [Wait] is called.
 func (f *Fetcher) Wait() error {
 	f.l.Lock()
-	if f.completed == f.numTxs {
-		close(f.fetchable)
-	}
+	close(f.fetchable)
```

Comment:
You don't need to close a channel in the lock, in fact I'm not sure if `f.l` is needed anymore